app/models/assignment_repo/creator.rb

```diff
@@ -189,5 +190,63 @@ def suffixed_repo_name(repository_name, suffix_count)
       suffix = "-#{suffix_count}"
       repository_name.truncate(100 - suffix.length, omission: '') + suffix
     end
+
+    # Handles assignment repository setup
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    #
+    # Returns nothing
+    def setup_repository(configs_repo, assignment_repo)
+      return unless configs_repo.classroom_configs?
+      conf_tree = configs_repo.branch('github-classroom').commit.commit.tree.sha
+      configs_repo.tree(conf_tree).tree.each do |conf|
+        process_config(configs_repo, assignment_repo, conf)
+      end
+
+      wait_import_completion assignment_repo
+
+      assignment_repo.remove_branch('github-classroom')
+    end
+
+    def wait_import_completion(github_repository)
+      t = 1
+      loop do
+        break if github_repository.import_progress[:status] == 'complete' || t > 60
+        sleep t += (t *= 0.25).ceil
+      end
+    end
+
+    # Process the configurations separetly
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    # conf            - GitObjects from of the 'github-classroom' branch tree
+    #                   of the configs_repo
+    #
+    # Returns nothing
+    def process_config(configs_repo, assignment_repo, conf)
+      case conf.path
+      when 'issues'
+        gen_issues(configs_repo, assignment_repo, conf.sha)
+      end
+    end
+
+    # Generates issues for the assignment_repo based on the configs
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    # issues_tree     - sha of the 'issues' tree
+    #
+    # Returns nothing
+    def gen_issues(configs_repo, assignment_repo, issues_tree)
+      configs_repo.tree(issues_tree).tree.each do |issue|
+        blob = configs_repo.blob(issue.sha)
+        assignment_repo.add_issue(blob.data['title'], blob.body)
+      end
+    end
```

Comment:
@nwoodthorpe I think this is fine to leave in here right now, and when we bring this to group assignment creation we can extract out duplicate logic then.